src/Synth/LFO.cpp

```diff
@@ -154,38 +162,34 @@ float LFO::biquad(float input)
     return (cutoff==127) ? input : output; // at cutoff 127 bypass filter
 }
 
-
 float LFO::lfoout()
 {
     //update internals XXX TODO cleanup
-    if ( ! lfopars_.time || lfopars_.last_update_timestamp == lfopars_.time->time())
+    if ( ! lfopars.time || lfopars.last_update_timestamp == lfopars.time->time())
     {
-        waveShape = lfopars_.PLFOtype;
-        int stretch = lfopars_.Pstretch;
-        if(stretch == 0)
-            stretch = 1;
-        const float lfostretch = powf(basefreq_ / 440.0f, (stretch - 64.0f) / 63.0f);
-
-        float lfofreq = lfopars_.freq * lfostretch;
-
-        phaseInc = fabsf(lfofreq) * dt_;
-
-        switch(lfopars_.fel) {
+        updatePars();
+        switch(lfopars.fel) {
             case consumer_location_type_t::amp:
-                lfointensity = lfopars_.Pintensity / 127.0f;
+                lfointensity = lfopars.Pintensity / 127.0f;
                 break;
             case consumer_location_type_t::filter:
-                lfointensity = lfopars_.Pintensity / 127.0f * 4.0f;
+                lfointensity = lfopars.Pintensity / 127.0f * 4.0f;
                 break; //in octave
             case consumer_location_type_t::freq:
             case consumer_location_type_t::unspecified:
-                lfointensity = powf(2, lfopars_.Pintensity / 127.0f * 11.0f) - 1.0f; //in centi
+                lfointensity = powf(2, lfopars.Pintensity / 127.0f * 11.0f) - 1.0f; //in centi
                 //x -= 0.25f; //chance the starting phase
                 break;
         }
     }
+    
+    if (lfopars.numerator && lfopars.denominator && tempo != time.tempo) {
+        tempo = time.tempo;
+        float lfofreq = (float(tempo)) / 15.0f * lfopars.denominator/lfopars.numerator;
+        phaseInc = fabsf(lfofreq) * dt;
+    }
 
-    float out = baseOut(waveShape, phase);
+    float out = baseOut(waveShape, fmod(phase + (lfopars.Pstartphase + 63.0f) / 127.0f, 1.0f));
```

Comment:
Allowing the start phase to change while the LFO is running is actually a feature which is independent to BPM sync?

Reply:
yes. but for synced LFOs it is more severely needed.